libs/angular-components/community/components/form/select/select-option.component.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  Component,
+  ElementRef,
+  computed,
+  inject,
+  input,
+  AfterContentInit,
+} from "@angular/core";
+import { CdkMenuModule, MenuStack, MENU_STACK } from "@angular/cdk/menu";
+import { SelectComponent } from "./select.component";
+import { DropdownItemComponent } from "./dropdown-item.component";
+
+@Component({
+  selector: "tedi-select-option",
+  standalone: true,
+  imports: [CdkMenuModule, DropdownItemComponent],
+  providers: [{ provide: MENU_STACK, useClass: MenuStack }],
+  template: `
+    <div
+      cdkMenuItem
+      tedi-dropdown-item
+      role="option"
+      (cdkMenuItemTriggered)="select()"
+      [selected]="isSelected()"
+      [disabled]="isDisabled()"
+      [attr.tabindex]="isDisabled() ? -1 : 0"
+      [attr.aria-selected]="isSelected()"
+      [attr.aria-disabled]="isDisabled()"
+    >
+      <ng-content />
+    </div>
+  `,
+})
+export class SelectOptionComponent implements AfterContentInit {
+  /*
+   * The value of the option.
+   */
+  value = input.required<any>();
+
+  /*
+   * Should the option be disabled?
+   */
+  isDisabled = input<boolean>(false);
+
+  private parent = inject(SelectComponent);
+  private elementRef = inject(ElementRef);
+  contentText: string = "";
+
+  isSelected = computed(() => {
+    return this.parent._selectedValue() === this.value();
+  });
+
+  ngAfterContentInit() {
+    // Extract text content from the component to use as label
+    this.contentText = this.elementRef.nativeElement.textContent.trim();
```

Comment:
maybe not necessary to store it as a class property, just extract it when selecting

Reply:
also check maybe is possible to have it just as a computed signal in the parent? from the SelectOptionComponent contentchildren filter out the selected option and show its label, without explicitly setting it